File: python/rules/behavior-and-failure-handling/no-fallback-return-in-except.py
def fetch_config():
    # ruleid: python-no-fallback-return-in-except
    try:
        val = int("invalid")
    except ValueError:
        return {}

def check_status():
    # ruleid: python-no-fallback-return-in-except
    try:
        val = int("invalid")
    except Exception:
        return False

def safe_handling():
    # ok: python-no-fallback-return-in-except
    try:
        val = int("123")
        return val
    except ValueError as exc:
        raise RuntimeError("Parsing failed") from exc

File: csharp/rules/behavior-and-failure-handling/no-fallback-return-in-except.cs
using System;

public class FallbackHandler
{
    public string FetchConfig()
    {
        // ruleid: csharp-no-fallback-return-in-except
        try
        {
            return int.Parse("invalid").ToString();
        }
        catch (FormatException ex)
        {
            return "default";
        }
    }

    public string SafeHandler()
    {
        // ok: csharp-no-fallback-return-in-except
        try
        {
            return int.Parse("123").ToString();
        }
        catch (FormatException ex)
        {
            throw new InvalidOperationException("Failed", ex);
        }
    }
}
